packages/riipen-ui/src/components/InputLabel.jsx

```diff
@@ -3,123 +3,105 @@ import PropTypes from "prop-types";
-  };
-
-  static defaultProps = {
-    classes: [],
-    color: "default",
-    required: false,
-    weight: "regular"
-  };
-
-  static contextType = ThemeContext;
-
-  render() {
-    const {
-      children,
-      color,
-      classes,
-      hint,
-      required,
-      weight,
-      ...other
-    } = this.props;
-
-    const theme = this.context;
-
-    const className = clsx(classes, color, weight);
-
-    const marginBottom = hint ? 1 : 3;
-
-    return (
-      <React.Fragment>
-        {children && (
-          <label className={className} {...other}>
-            <Typography color="inherit" variant="inherit">
-              {children}
-              {required && " *"}
-            </Typography>
-          </label>
-        )}
-        {hint && <InputHint color={color}>{hint}</InputHint>}
-        <style jsx>{`
-          label {
-            color: ${theme.palette.text.secondary};
-            display: inline-block;
-            font-family: ${theme.typography.body1.fontFamily};
-            font-size: 16px;
-            font-weight: ${theme.typography.body1.fontWeight};
-            letter-spacing: ${theme.typography.body1.letterSpacing};
-            line-height: 1.4;
-            margin-bottom: ${theme.spacing(marginBottom)}px;
-          }
-
-          .white {
-            color: ${theme.palette.common.white};
-          }
-
-          .black {
-            color: ${theme.palette.common.black};
-          }
-
-          .grey {
-            color: ${theme.palette.grey.A400};
-          }
-
-          .light {
-            font-weight: ${theme.typography.fontWeight.light};
-          }
-
-          .medium {
-            font-weight: ${theme.typography.fontWeight.medium};
-          }
-
-          .bold {
-            font-weight: ${theme.typography.fontWeight.bold};
-          }
-        `}</style>
-      </React.Fragment>
-    );
-  }
-}
-
-export default withClasses(InputLabel);
+const InputLabel = ({
+  children,
+  color,
+  classes,
+  fontWeight,
+  hint,
+  required,
+  theme,
+  variant,
+  ...other
+}) => {
+  const className = clsx(classes);
+
+  const marginBottom = hint ? 1 : 3;
+
+  return (
+    <React.Fragment>
+      {children && (
+        <label className={className} {...other}>
+          <Typography color={color} fontWeight={fontWeight} variant={variant}>
+            {children}
+            {required && " *"}
+          </Typography>
+        </label>
+      )}
+      {hint && <InputHint color={color}>{hint}</InputHint>}
+      <style jsx>{`
+        label {
+          color: ${theme.palette.text.secondary};
+          display: inline-block;
+          font-family: ${theme.typography.body1.fontFamily};
+          font-size: 16px;
+          font-weight: ${theme.typography.body1.fontWeight};
+          letter-spacing: ${theme.typography.body1.letterSpacing};
+          line-height: 1.4;
+          margin-bottom: ${theme.spacing(marginBottom)}px;
+        }
```

Comment:
We could move this into a typography variant. i left it here so we can leave the default props the way they are